src/Windowing/OpenToolkit.Windowing.Common/Input/KeyboardState.cs

```diff
@@ -17,99 +19,67 @@ namespace OpenToolkit.Windowing.Common.Input
     public struct KeyboardState : IEquatable<KeyboardState>
     {
         // Allocate enough ints to store all keyboard keys
-        private const int IntSize = sizeof(int) * 8;
+        private const int ByteSize = 8;
 
-        private const int NumInts = ((int)Key.LastKey + IntSize - 1) / IntSize;
+        private const int NumBytes = ((int)Key.LastKey + ByteSize) / ByteSize;
 
-        // The following line triggers bogus CS0214 in gmcs 2.0.1, sigh...
-        private unsafe fixed int _keys[NumInts];
+        private unsafe fixed int _keys[NumBytes];
 
         /// <summary>
         /// Gets a <see cref="bool" /> indicating whether the specified
-        ///  <see cref="Key" /> is pressed.
+        ///  <see cref="Key" /> is currently down.
         /// </summary>
         /// <param name="key">The <see cref="Key" /> to check.</param>
-        /// <returns><c>true</c> if key is pressed; <c>false</c> otherwise.</returns>
+        /// <returns><c>true</c> if key is down; <c>false</c> otherwise.</returns>
         public bool this[Key key]
         {
             get => IsKeyDown(key);
-            internal set => SetKeyState(key, value);
+            set => SetKeyState(key, value);
         }
 
         /// <summary>
-        /// Gets a <see cref="bool" /> indicating whether the specified
-        ///  <see cref="OpenToolkit.Windowing.Common.Input.Key" /> is pressed.
-        /// </summary>
-        /// <param name="code">The scancode to check.</param>
-        /// <returns><c>true</c> if code is pressed; <c>false</c> otherwise.</returns>
-        public bool this[short code] => IsKeyDown((Key)code);
-
-        /// <summary>
-        /// Gets a <see cref="bool" /> indicating whether this key is down.
+        /// Gets a <see cref="bool" /> indicating whether this key is currently down.
         /// </summary>
         /// <param name="key">The <see cref="Key" /> to check.</param>
         /// <returns><c>true</c> if <paramref name="key"/> is in the down state; otherwise, <c>false</c>.</returns>
-        public bool IsKeyDown(Key key)
+        public unsafe bool IsKeyDown(Key key)
         {
-            return ReadBit((int)key);
-        }
+            if (key <= Key.Unknown || key > Key.LastKey)
+            {
+                throw new ArgumentOutOfRangeException(nameof(key), "Invalid key");
+            }
 
-        /// <summary>
-        /// Gets a <see cref="bool" /> indicating whether this scan code is down.
-        /// </summary>
-        /// <param name="code">The scan code to check.</param>
-        /// <returns>
-        /// <c>true</c> if the key given by the <paramref name="code"/> parameter is in the down state;
-        /// otherwise, <c>false</c>.
-        /// </returns>
-        public bool IsKeyDown(short code)
-        {
-            return code >= 0 && code < (short)Key.LastKey && ReadBit(code);
+            var offset = (int)key;
+            ValidateOffset(offset);
+
+            var intOffset = offset / ByteSize;
```

Comment:
only correct if a byte array would be used - see above(which I would not recommend)

Reply:
Damnit, I DID intend to use a byte array but I guess while testing I messed it up and put it back to an int array. Good catch.